src/demo/apps/LauncherApp.tsx

```diff
@@ -62,30 +95,55 @@ export function LauncherApp(): React.ReactElement {
+            <hr/>
+            <h2>Programmatic apps</h2>
+            {NON_MANIFEST_APPS.map((app, index) => <AppCard key={app.data.uuid + index} app={app} handleClick={launchApp} isDirectoryApp={false} />)}
         </div>
     );
 }
 
-const NON_DIRECTORY_APPS: Application[] = ([
-    {id: 'blotter', icon: 'blotter', title: 'Blotter', description: 'Sample non-directory blotter app'},
-    {id: 'contacts', icon: 'contacts', title: 'Contacts', description: 'Sample non-directory contacts app'},
-    {id: 'dialer', icon: 'dialer', title: 'Dialer', description: 'Sample non-directory dialer app'},
-    {id: 'charts-pink', icon: 'charts', title: 'Charts: Pink', description: 'A non-directory charting app'},
-    {id: 'charts-grey', icon: 'charts', title: 'Charts: Grey', description: 'Another non-directory charting app'},
-    {id: 'charts-teal', icon: 'charts', title: 'Charts: Teal', description: 'Another non-directory charting app'},
-    {id: 'news', icon: 'news', title: 'News Feed', description: 'Sample non-directory news app'}
-] as Array<{id: string, icon: string, title: string, description: string}>).map(({id, icon, title, description}) => ({
-    appId: `fdc3-${id}-nodir`,
-    name: `fdc3-${id}-nodir`,
-    manifestType: 'openfin',
-    manifest: `http://localhost:3923/demo/configs/non-directory/app-${id}-nodir.json`,
-    icons: [
-        {icon: `http://localhost:3923/demo/img/app-icons/${icon}.svg`}
-    ],
-    title: title || id,
-    description
+const APP_DATA: Array<AppData> = [
+    {id: 'blotter', icon: 'blotter', title: 'Blotter', description: 'blotter app'},
+    {id: 'contacts', icon: 'contacts', title: 'Contacts', description: 'contacts app'},
+    {id: 'dialer', icon: 'dialer', title: 'Dialer', description: 'dialer app',
+        extraOptions: {resizable: false, defaultWidth: 240, defaultHeight: 310, saveWindowState: false}},
+    {id: 'charts-pink', icon: 'charts', title: 'Charts: Pink', description: 'charting app'},
+    {id: 'charts-grey', icon: 'charts', title: 'Charts: Grey', description: 'charting app'},
+    {id: 'charts-teal', icon: 'charts', title: 'Charts: Teal', description: 'charting app'},
+    {id: 'news', icon: 'news', title: 'News Feed', description: 'news app'}
+];
+
+const NON_DIRECTORY_APPS: ManifestAppLaunchData[] = APP_DATA.map(({id, icon, title, description}) => ({
+    type: 'manifest',
+    data: {
+        appId: `fdc3-${id}-nodir`,
+        name: `fdc3-${id}-nodir`,
+        manifestType: 'openfin',
+        manifest: `http://localhost:3923/demo/configs/non-directory/app-${id}-nodir.json`,
+        icons: [
+            {icon: `http://localhost:3923/demo/img/app-icons/${icon}.svg`}
+        ],
+        title: title || id,
+        description: 'Sample Non-directory ' + description
+    }
+}));
+
+const NON_MANIFEST_APPS: ProgrammaticAppLaunchData[] = APP_DATA.map(({id, icon, title, description, extraOptions}) => ({
```

Comment:
The charting apps should use different colours to the non directory apps, you can probably just do a `.replace(a, b).replace(c,d)...`.

The colours come from css files `https://www.w3schools.com/lib/w3-theme-yellow.css` and I believe they are just generated from the css color names, [go wild](https://www.w3schools.com/colors/colors_names.asp) 😄

Reply:
@TheoOttahOF That one is done in AppCard.tsx:26 already. and went wild with w3-light-green :P

Reply:
this has changed in scope of AppCard review changes.